src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -86,52 +86,36 @@ export const portfolio = createSlice({
 
       Object.assign(state.connectedWallet, { supportedChainIds: payload })
     },
-    upsertAccountMetadata: {
-      reducer: (
-        draftState,
-        {
-          payload,
-        }: { payload: { accountMetadataByAccountId: AccountMetadataById; walletId: string } },
-      ) => {
-        // WARNING: don't use the current state.connectedWallet.id here because it's updated async
-        // to this and results in account data corruption
-        const { accountMetadataByAccountId, walletId } = payload
-        draftState.accountMetadata.byId = merge(
-          draftState.accountMetadata.byId,
-          accountMetadataByAccountId,
-        )
-        draftState.accountMetadata.ids = Object.keys(draftState.accountMetadata.byId)
-
-        if (!draftState.connectedWallet) return // realistically, at this point, we should have a wallet set
-        const existingWalletAccountIds = draftState.wallet.byId[walletId] ?? []
-        const newWalletAccountIds = Object.keys(accountMetadataByAccountId)
-        // keep an index of what account ids belong to this wallet
-        draftState.wallet.byId[walletId] = uniq(
-          existingWalletAccountIds.concat(newWalletAccountIds),
-        )
-      },
+    upsertAccountMetadata: (
```

Comment:
Review without whitespaces diff, remove `prepare` so that account meta upsertion isn't a low prio action anymore, we need this as a high prio